tests/constructors/provider_test.py

```diff
@@ -119,11 +119,19 @@ def local_function(): ...
         Provider(local_function)
 
 
-def func_with_union_arg(_: Union[int, float, None]) -> int:
+def func_with_union_arg(_: float | str | None) -> int | str:
     return 0
 
 
-def func_with_union_return() -> Union[int, float, None]:
+def func_with_explicit_union_arg(_: Union[float, str, None]) -> int | str:
+    return 0
+
+
+def func_with_union_return() -> int | str | None:
+    return None
+
+
+def func_with_explicit_union_return() -> Union[int, str, None]:
```

Comment:
Does this really make a difference compared to the pipe syntax? 

In any case, you should suppress the relevant `UP` rule here.

Reply:
Yes, if you make Union with `Union`, the `get_origin` returns `Union` but if you make it with the pipe syntax, `get_origin` returns `UnionType`. 

```python
>>> from typing import get_origin, Union 
>>> get_origin(int | str)
<class 'types.UnionType'>
>>> get_origin(Union[int, str])
typing.Union
```

What do you mean by the `UP` rule though?

Reply:
PYI041 Use float instead of int | float
This one you mean?

Reply:
No, I meant this one: https://docs.astral.sh/ruff/rules/non-pep604-annotation/ It is not enabled in our setup but I used the pyupgrade rules to upgrade the code in other repos. Suppressing this here ensures that we don't accidentally change the function in the future.

Reply:
Apparently it's not allowed to suppress unused rule!
`RUF100 Unused `noqa` directive (non-enabled: `UP007`)`
I'll just enable `UP` and fix it there...! Thanks : D

Reply:
You would also have to suppress `RUF100` in the same line.

You are welcome to enable `UP`. I didn't do that in the template because it does not always lead to desirable code. E.g., https://docs.astral.sh/ruff/rules/non-pep604-isinstance/ reduces performance. So I would at least not enable auto fixes for all `UP` rules.

Reply:
Done in https://github.com/scipp/beamlime/blob/fba4b3ddf17a235ae73adc8f696d37785e6a2c8a/tests/constructors/factory_test.py#L66-L77
and
https://github.com/scipp/beamlime/blob/fba4b3ddf17a235ae73adc8f696d37785e6a2c8a/tests/constructors/factory_test.py#L120-L133